### PR TITLE
GetAppServiceRecords RPC ATF Tests

### PR DESCRIPTION
**Pull request by ShobhitAd**
This PR is **ready** for review.

### Summary
Test scripts to test the GetAppServiceRecords RPC

### ATF version
[feature/app_services branch](https://github.com/smartdevicelink/sdl_atf/tree/feature/app_services)

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: test_scripts/AppServices/GetAppServiceRecords/001_GetAppServiceRecords_MOBILE.lua [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+local runner = require('user_modules/script_runner')
+local common = require('test_scripts/AppServices/commonAppServices')
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+--[[ Local functions ]]
+local function PTUfunc(tbl)
+    local pt_entry = common.getAppServiceProducerConfig(1)
+    pt_entry.app_services.MEDIA = { handled_rpcs = {{function_id = 41}} }
```
Comment (JackLivio) [inline test_scripts/AppServices/GetAppServiceRecords/001_GetAppServiceRecords_MOBILE.lua]: Did you mean to overwrite the service name here?

Comment (JackLivio) [reply test_scripts/AppServices/GetAppServiceRecords/001_GetAppServiceRecords_MOBILE.lua]: pt_entry.app_services.MEDIA.handled_rpcs = {{function_id = 41}} ?

File: test_scripts/AppServices/GetAppServiceRecords/003_GetAppServiceRecords_Multiple.lua [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+--[[ Local functions ]]
+local function PTUfunc(tbl)
+    local pt_entry = common.getAppServiceProducerConfig(1)
+    pt_entry.app_services.MEDIA = { handled_rpcs = {{function_id = 41}} }
+    tbl.policy_table.app_policies[common.getConfigAppParams(1).fullAppID] = pt_entry
+end
+
+local function getAppServiceRecords(serviceType, expectedResponse)
+    expectedResponse = expectedResponse()
```
Comment (JackLivio) [inline test_scripts/AppServices/GetAppServiceRecords/003_GetAppServiceRecords_Multiple.lua]: No need to pass expectedResponse as a parameter when you just call the local function anyways

File: test_scripts/AppServices/GetAppServiceRecords/002_GetAppServiceRecords_HMI.lua [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+---------------------------------------------------------------------------------------------------
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local common = require('test_scripts/AppServices/commonAppServices')
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+--[[ Local functions ]]
+local function getAppServiceRecords(serviceType, expectedResponse)
```
Comment (JackLivio) [inline test_scripts/AppServices/GetAppServiceRecords/002_GetAppServiceRecords_HMI.lua]: No need to pass expectedResponse as a parameter when you just call the local function anyways

File: test_scripts/AppServices/GetAppServiceRecords/001_GetAppServiceRecords_MOBILE.lua [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    tbl.policy_table.app_policies[common.getConfigAppParams(1).fullAppID] = pt_entry
+end
+
+local function getAppServiceRecords(serviceType, expectedResponse)
+    expectedResponse = expectedResponse()
```
Comment (JackLivio) [inline test_scripts/AppServices/GetAppServiceRecords/001_GetAppServiceRecords_MOBILE.lua]: No need to pass expectedResponse as a parameter when you just call the local function anyways

File: test_scripts/AppServices/GetAppServiceRecords/001_GetAppServiceRecords_MOBILE.lua (hunk 1 above)
Comment (ShobhitAd) [reply test_scripts/AppServices/GetAppServiceRecords/001_GetAppServiceRecords_MOBILE.lua]: whoops... not intentional. Carried over my changes from RPC passing where I had to define a new NAVIGATION section. I'll commit this fix

**Pull request merged**